### PR TITLE
Adding callback to artoo.beep

### PR DESCRIPTION
**Pull request by aileo**
can now be called as follow :
```js
artoo.beep();
artoo.beep(sound);
artoo.beep(sound, callback);
artoo.beep(callback, sound);
artoo.beep(callback);
```

### CODE REVIEW

**Comment by Yomguithereal** [comment]
Hello @aileo. Thanks for this PR representing new exciting step towards the creation of majestuous beeping symphonies.

I however think that the `artoo.beep(callback, sound)` polymorphism might be overkill and misleading. Could remove it please?

File: src/artoo.beep.js [hunk 1]
```diff
@@ -26,8 +26,22 @@
...
+
+    if (typeof a1 === 'function') {
+      callback = a1;
+    }
+    else {
+      sound = a1;
+      if (typeof a2 === 'function') 
+        callback = a2;
+      else
+        throw Error('artoo.beep: second argument have to be a function.');
```
Comment (Yomguithereal) [inline src/artoo.beep.js]: The `beep` method cannot be called without callback. This is problematic, my liege.

**Comment by aileo** [comment]
Please don't merge it before I rebase as one clean commit, assuming this time to be the good one.

**Comment by Yomguithereal** [comment]
Too late darling.

**Pull request merged**